Presentation/PresentationOptions.swift

```diff
@@ -46,6 +46,10 @@ public extension PresentationOptions {
 
     /// Default options used unless any options are explicity passed when presented: `[embedInNavigationController]`
     static let defaults: PresentationOptions = [embedInNavigationController]
+
+    static func prefersNavigationBarHidden(_ hidden: Bool) -> PresentationOptions {
+        return hidden ? navigationBarHidden : navigationBarShown
+    }
```

Comment:
We discussed over slack if it was possible to just have one option `. prefersNavigationBarHidden`. I think that would be much more elegant. And as for backward compatibility, I suggested that the we keep the behavior we have now until a view controller is first pushed with `.prefersNavigationBarHidden`, where the user has stated its intention to use Presentation FW's handling navigation bar visibility.

So if the current push popper stack does not include any `.prefersNavigationBarHidden` disable navigation bar logic. But once a view controller is push with `.prefersNavigationBarHidden` enable logic.

So if the new top controller after processPushPopers has `.prefersNavigationBarHidden`, hide the navigation bar. And if the new top controller don't have `.prefersNavigationBarHidden` and any previous push popper (either just pop:ed or push:ed before), show the navigation bar. If none of these are true don't do anything with the navigation bar visibility (backward compatibility mode).

Do think this could work?

Reply:
@mansbernhardt  Yes. I remember we talked about this. And when I tried to imagine all the use cases. I don't think, only having `.prefersNavigationBarHidden` as option suits the scenario below:

`navigationController(nav bar is hidden) -> vc1 -> vc2  -> vc3 (wants NavBar) -> vc4 (no Navbar) -> vc5 (no navbar)`

You cannot specify any option to show the navigationBar for VC3 in presentation option, if we only have `.prefersNavigationBarHidden`.

Thus allowing to either specify the preference or to not specify anything that will keep existing state makes more sense to me.

Please let me know your thoughts on this @nataliq